release/releaser_test.go

```diff
@@ -40,157 +110,282 @@ func setup(t *testing.T, mocks instance.Instance) (*Releaser, func()) {
+					Automated: false,
+					Locked:    true,
+				},
+			},
 		},
-	}
-	if !reflect.DeepEqual(expected, results) {
-		t.Errorf("expected %#v, got %#v", expected, results)
+		Error: nil,
 	}
 
-	spec = jobs.ReleaseJobParams{
-		ReleaseSpec: flux.ReleaseSpec{
-			ServiceSpec: flux.ServiceSpec("default/helloworld"),
-			ImageSpec:   flux.ImageSpecLatest,
-			Kind:        flux.ReleaseKindPlan,
+	for _, tst := range []struct {
+		Name     string
+		Spec     flux.ReleaseSpec
+		Expected flux.ReleaseResult
+	}{
+		// ignored if: excluded OR not included OR not correct image.
+		{
+			Name: "not included",
+			Spec: flux.ReleaseSpec{
+				ServiceSpec: hwSvcSpec,
+				ImageSpec:   flux.ImageSpecLatest,
+				Kind:        flux.ReleaseKindExecute,
+				Excludes:    []flux.ServiceID{},
+			},
+			Expected: flux.ReleaseResult{
+				flux.ServiceID("default/helloworld"): flux.ServiceResult{
+					Status: flux.ReleaseStatusSuccess,
+					PerContainer: []flux.ContainerUpdate{
+						flux.ContainerUpdate{
+							Container: "helloworld",
+							Current:   oldImageID,
+							Target:    newImageID,
+						},
+					},
+				},
+				flux.ServiceID("default/locked-service"): flux.ServiceResult{
+					Status: flux.ReleaseStatusIgnored,
+					Error:  NotIncluded,
+				},
+				flux.ServiceID("default/test-service"): flux.ServiceResult{
+					Status: flux.ReleaseStatusIgnored,
+					Error:  NotIncluded,
+				},
+			},
+		}, {
+			Name: "excluded",
+			Spec: flux.ReleaseSpec{
+				ServiceSpec: flux.ServiceSpecAll,
+				ImageSpec:   flux.ImageSpecLatest,
+				Kind:        flux.ReleaseKindExecute,
+				Excludes:    []flux.ServiceID{lockedSvcID},
+			},
+			Expected: flux.ReleaseResult{
+				flux.ServiceID("default/helloworld"): flux.ServiceResult{
+					Status: flux.ReleaseStatusSuccess,
+					PerContainer: []flux.ContainerUpdate{
+						flux.ContainerUpdate{
+							Container: "helloworld",
+							Current:   oldImageID,
+							Target:    newImageID,
+						},
+					},
+				},
+				flux.ServiceID("default/locked-service"): flux.ServiceResult{
+					Status: flux.ReleaseStatusIgnored,
+					Error:  Excluded,
+				},
+				flux.ServiceID("default/test-service"): flux.ServiceResult{
+					Status: flux.ReleaseStatusSkipped,
+					Error:  NotInCluster,
+				},
+			},
+		}, {
+			Name: "not image",
+			Spec: flux.ReleaseSpec{
+				ServiceSpec: flux.ServiceSpecAll,
+				ImageSpec:   flux.ImageSpecFromID(newImageID),
+				Kind:        flux.ReleaseKindExecute,
+				Excludes:    []flux.ServiceID{},
+			},
+			Expected: flux.ReleaseResult{
+				flux.ServiceID("default/helloworld"): flux.ServiceResult{
+					Status: flux.ReleaseStatusSuccess,
+					PerContainer: []flux.ContainerUpdate{
+						flux.ContainerUpdate{
+							Container: "helloworld",
+							Current:   oldImageID,
+							Target:    newImageID,
+						},
+					},
+				},
+				flux.ServiceID("default/locked-service"): flux.ServiceResult{
+					Status: flux.ReleaseStatusIgnored,
+					Error:  DifferentImage,
```

Comment:
Isn't this test saying that for `--all` locked services for different images will still be output?